enhancements/machine-api/spot-instances.md

```diff
@@ -86,41 +86,122 @@ the following requirements for integration will work for each of AWS, Azure and
 #### Termination handler design
 
 To enable graceful termination of workloads running on non-guaranteed instances,
-three controllers will need to be implemented to watch for termination notices
-and gracefully move workloads
-
-##### Termination Pod DaemonSet
-
-The Termination Pod DaemonSet would be deployed will be deployed by the Machine API Operator and
-will select Nodes which are labelled as spot instances.
-This will ensure the Termination Pod only runs on instances that need it.
-
-The spot label can be added by the cloud provider actuators as they create instances,
-provided they support spot instances and the instance is a spot instance.
+a termination handler pod will need to be deployed on each spot instance.
+
+The termination pod will be developed to poll the metadata service for the Node
+that it is running on.
+We will implement request/response handlers for each of the three cloud providers
+that will enable the handler to determine if the cloud provider instance is due
+for termination.
+
+Should the instance be marked for termination, the handler will add a condition
+to the Node object that it is running on to signal that the instance will be
+terminating imminently.
+
+```yaml
+- type:     Terminating
+  status:   True
+  Reason:   TerminationRequested
+  Message:  The cloud provider has marked this instance for termination
+```
 
-##### Termination Pod
+Once the Node has been marked with the `Terminating` condition, it will be
+the MachineHealthCheck controller's responsibility to ensure that the Machine
```

Comment:
to clarify the MachineHealthCheck controller will issue a deletion request. The machine controller will drain by default.

Reply:
I've updated the sentence slightly, does that make it clearer?